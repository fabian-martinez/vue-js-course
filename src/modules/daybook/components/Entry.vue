<template lang="">
    <div class="entry-container mb-3 pointer p-2" @click="$router.push( { name: 'entry', params: { id:entry.id }   })">
        <!-- Titulo -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">{{day}}</span>
            <span class="mx-1 fs-5">{{month}}</span>
            <span class="mx-2 fw-light">{{year}}, {{weekDay}}</span>
        </div>
         <!-- DescripciÃ³n -->
         <div class="entry-description">
            {{ shortText }}
         </div>
    </div>
</template>
<script>
import getDayMonthYear from '@/modules/daybook/helpers/getDayMonthYear'
export default {
    props: {
        entry:{
            type:Object,
            required:true
        }
    },
    computed:{
        shortText(){
             return ( this.entry.text.lenght > 130 )
                ? this.entry.text.substring(0,130) + '...'
                : this.entry.text
        },
        weekDay(){
            const { weekDay } = getDayMonthYear( this.entry.date )
            return weekDay
        },
        day(){
            const { day } = getDayMonthYear( this.entry.date )
            return day
        },
        month(){
            const { month } = getDayMonthYear( this.entry.date )
            return month
        },
        year(){
            const { year } = getDayMonthYear( this.entry.date )
            return year
        }

    },
}
</script>
<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;
    &:hover{
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
    .entry-description {
        font-size: 12ps;
    }
}
</style>