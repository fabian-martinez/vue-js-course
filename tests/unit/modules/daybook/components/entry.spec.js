import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";


describe('Pruebas en el componente Entry', () => {

    const mockRouter = { push: jest.fn() }
    const wrapper = shallowMount(Entry, {
        props: { entry: { ...journalState.entries[0] }},
        global:{ mocks: { $router: mockRouter }}
    }) 

    it('debe hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('deber redireccionar al hacer click en el entry-container', () => {

        const entryThunmail = wrapper.find('.entry-container')

        entryThunmail.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalled()
        expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'entry', params: {id:journalState.entries[0].id} })

    });

    it('pruebas en las propiedades computadas', () => {

        expect(wrapper.vm.weekDay).toEqual('domingo')
        expect(wrapper.vm.day).toEqual(6);
        expect(wrapper.vm.month).toEqual('noviembre')
        expect(wrapper.vm.year).toEqual(2022)
        // wrapper.vm. <--- popiedades computadas 
    });

});