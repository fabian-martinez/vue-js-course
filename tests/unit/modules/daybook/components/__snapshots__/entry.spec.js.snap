// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Entry debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">6</span><span class="mx-1 fs-5">noviembre</span><span class="mx-2 fw-light">2022, domingo</span></div><!-- DescripciÃ³n -->
  <div class="entry-description">Entrada para pruebas unitarias enfrenase</div>
</div>
`;
